pkg/kube/services/services.go

```diff
@@ -92,7 +95,43 @@ func GetServiceURLFromMap(services map[string]*v1.Service, name string) string {
 	return GetServiceURL(services[name])
 }
 
-func FindServiceURL(client kubernetes.Interface, namespace string, name string) (string, error) {
+func getIstioVirtualService(dynamicClient dynamic.Interface, namespace, name string) (*unstructured.Unstructured, error) {
+	dynamicClient, err := kube.LazyCreateDynamicClient(dynamicClient)
+	if err != nil {
+		return nil, err
+	}
+	//  Create a GVR which represents an Istio Virtual Service.
+	virtualServiceGVR := schema.GroupVersionResource{
+		Group:    "networking.istio.io",
+		Version:  "v1alpha3",
+		Resource: "virtualservices",
+	}
+	virtualService, err := dynamicClient.Resource(virtualServiceGVR).Namespace(namespace).Get(context.TODO(), name, meta_v1.GetOptions{})
+	return virtualService, err
+}
+
+// getUrlFromVirtualService finds url from virtual services istio
+func getUrlFromVirtualService(virtualService *unstructured.Unstructured) (string, error) {
+	vs := virtualService.Object
+	if spec, ok := vs["spec"].(map[string]interface{}); ok {
+		if hosts, ok := spec["hosts"].([]interface{}); ok {
+			return fmt.Sprintf("%v", hosts[0]), nil
+		}
+	}
+	return "", errors.New("No url found in the virtual service")
+}
+
+// FindUrlFromVsIstio finds the host from istio virtual service
+func FindUrlFromVsIstio(dynamicClient dynamic.Interface, namespace, name string) (string, error) {
+	virtualService, err := getIstioVirtualService(dynamicClient, namespace, name)
+	if err != nil {
```

Comment:
getIstioVirtualService will return error if there is no istio in that case .Should i do it or 
_, err = client.Discovery().ServerResourcesForGroupVersion("networking.istio.io/v1alpha3")
if err == nil {
// Istio specific code
}
In this i feel like doing the same operation twice if istio is present in that service.

Reply:
> getIstioVirtualService will return error if there is no istio in that case .

Of course. The question is whether you can distinguish whether the error is due to istio not being installed or not.

Reply:
yes this will do then.